core/input/mouse.js

```diff
@@ -108,27 +117,82 @@ Mouse.prototype = {
         this._handleMouseButton(e, 0);
     },
 
+    // Mouse wheel events are sent in steps over VNC. This means that the VNC
+    // protocol can't handle a wheel event with specific distance or speed.
+    // Therefor, if we get a lot of small mouse wheel events we combine them.
+    _generateWheelStepX: function () {
+
+        if (this._accumulatedWheelDeltaX < 0) {
+            this._onMouseButton(this._wheelPos.x, this._wheelPos.y, 1, 1 << 5);
+            this._onMouseButton(this._wheelPos.x, this._wheelPos.y, 0, 1 << 5);
+        } else if (this._accumulatedWheelDeltaX > 0) {
+            this._onMouseButton(this._wheelPos.x, this._wheelPos.y, 1, 1 << 6);
+            this._onMouseButton(this._wheelPos.x, this._wheelPos.y, 0, 1 << 6);
+        }
+
+        this._accumulatedWheelDeltaX = 0;
+    },
+
+    _generateWheelStepY: function () {
+
+        if (this._accumulatedWheelDeltaY < 0) {
+            this._onMouseButton(this._wheelPos.x, this._wheelPos.y, 1, 1 << 3);
+            this._onMouseButton(this._wheelPos.x, this._wheelPos.y, 0, 1 << 3);
+        } else if (this._accumulatedWheelDeltaY > 0) {
+            this._onMouseButton(this._wheelPos.x, this._wheelPos.y, 1, 1 << 4);
+            this._onMouseButton(this._wheelPos.x, this._wheelPos.y, 0, 1 << 4);
+        }
+
+        this._accumulatedWheelDeltaY = 0;
+    },
+
+    _resetWheelStepTimers: function () {
+        window.clearTimeout(this._wheelStepXTimer);
+        window.clearTimeout(this._wheelStepYTimer);
+        this._wheelStepXTimer = null;
+        this._wheelStepYTimer = null;
+    },
+
     _handleMouseWheel: function (e) {
-        if (!this._focused) { return; }
+        if (!this._focused || !this._onMouseButton) { return; }
 
-        var pos = this._getMousePosition(e);
+        this._resetWheelStepTimers();
 
-        if (this._onMouseButton) {
-            if (e.deltaX < 0) {
-                this._onMouseButton(pos.x, pos.y, 1, 1 << 5);
-                this._onMouseButton(pos.x, pos.y, 0, 1 << 5);
-            } else if (e.deltaX > 0) {
-                this._onMouseButton(pos.x, pos.y, 1, 1 << 6);
-                this._onMouseButton(pos.x, pos.y, 0, 1 << 6);
-            }
+        this._wheelPos = this._getMousePosition(e);
```

Comment:
This doesn't really solve the position problem as there might be a normal movement of the mouse before the timer fires. You should keep track of the last known position, no matter what event it was.

(and the naming should probably reflect that as well :))

Reply:
Right, so basically update the position in all mouse handlers?

Reply:
Indeed. Perhaps the helper can be changed for this?

Reply:
how do you mean?

Reply:
How do you mean?

Reply:
We have `_getMousePosition()`. Perhaps it should be changed to something like `_updateMousePosition()`?

Reply:
right, of course